app/src/main/java/br/com/orcinus/orca/app/activity/masking/Masker.kt

```diff
@@ -0,0 +1,219 @@
+ * that is, those which ought to be rounded according to the design specification of Orca.
+ *
+ * @see HardwareRoundedCorners
+ */
+enum class Masker {
+  /** Masks the trailing corner at the bottom of the [MaskableFrameLayout]. */
+  BOTTOM_END {
+    override fun getCornerRadiiIndices(layoutDirection: Int): IntArray {
+      return if (layoutDirection == View.LAYOUT_DIRECTION_RTL) {
+        bottomLeftRadiiIndices
+      } else {
+        bottomRightRadiiIndices
+      }
+    }
+
+    override fun getRoundedCornerRadius(
+      hardwareRoundedCorners: HardwareRoundedCorners,
+      layoutDirection: Int
+    ): Float {
+      return if (layoutDirection == View.LAYOUT_DIRECTION_RTL) {
+        hardwareRoundedCorners.bottomLeft()
+      } else {
+        hardwareRoundedCorners.bottomRight()
+      }
+    }
+
+    override fun getDefaultRadius(displayMetrics: DisplayMetrics): Float {
+      return TypedValue.applyDimension(
+        TypedValue.COMPLEX_UNIT_DIP,
+        defaultForm.bottomEnd,
+        displayMetrics
+      )
+    }
+  },
+
+  /** Masks the leading portion at the bottom of the [MaskableFrameLayout]. */
+  BOTTOM_START {
+    override fun getCornerRadiiIndices(layoutDirection: Int): IntArray {
+      return if (layoutDirection == View.LAYOUT_DIRECTION_RTL) {
+        bottomRightRadiiIndices
+      } else {
+        bottomLeftRadiiIndices
+      }
+    }
+
+    override fun getRoundedCornerRadius(
+      hardwareRoundedCorners: HardwareRoundedCorners,
+      layoutDirection: Int
+    ): Float {
+      return if (layoutDirection == View.LAYOUT_DIRECTION_RTL) {
+        hardwareRoundedCorners.bottomRight()
+      } else {
+        hardwareRoundedCorners.bottomLeft()
+      }
+    }
+
+    override fun getDefaultRadius(displayMetrics: DisplayMetrics): Float {
+      return TypedValue.applyDimension(
+        TypedValue.COMPLEX_UNIT_DIP,
+        defaultForm.bottomStart,
+        displayMetrics
+      )
+    }
+  };
+
+  /**
+   * Obtains the indices at which the radii for the equivalent corner are, as defined by
+   * [GradientDrawable].
+   *
+   * @param layoutDirection Integer that represents the current direction of the layout.
+   *   Left-to-right (LTR) is denoted by [View.LAYOUT_DIRECTION_LTR]; right-to-left (RTL),
+   *   [View.LAYOUT_DIRECTION_RTL]. If the actual value is neither of those, this method will apply
+   *   the radius it would for LTR.
+   */
+  protected abstract fun getCornerRadiiIndices(layoutDirection: Int): IntArray
+
+  /**
+   * Obtains the radius in pixels of the equivalent corner of the display, defaulting to that of
+   * αὐτός' large form in case it is not provided or the current API level is an unsupported one —
+   * that is, < 31 (S).
+   *
+   * @param view [View] to which the mask is to be applied.
+   * @see Forms.large
+   */
+  protected fun getRadius(hardwareRoundedCorners: HardwareRoundedCorners, view: View): Float {
+    return (null as Float?)
```

Comment:
Maybe [`Float.NaN`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-float/-na-n.html) to avoid boxing?